packages/components/src/components/amount-input/amount-input.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useCallback, useState } from 'react';
+import { isMobile } from '@deriv/shared';
+import Input from '../input';
+import Text from '../text';
+
+type TAmountInput = {
+    currency: string;
+    decimal_places?: number;
+    disabled?: boolean;
+    initial_value?: number;
+    label?: string;
+    max_digits?: number;
+    onChange?: (value: number) => void;
+};
+
+const AmountInput = ({
+    currency,
+    decimal_places = 2,
+    disabled = false,
+    initial_value = 0,
+    label,
+    max_digits = 8,
+    onChange,
+}: TAmountInput) => {
+    const [value, setValue] = useState(initial_value);
+    const [focus, setFocus] = useState(false);
+
+    const displayNumber = useCallback(
+        (number: number) => number.toLocaleString(undefined, { minimumFractionDigits: decimal_places }),
+        [decimal_places]
+    );
+
+    const onChangeHandler = (e: { target: { value: string } }) => {
+        const input_value = e.target.value.replace(/\D/g, '');
+        if (Number(input_value) <= Math.pow(10, max_digits)) {
+            setValue(Number(input_value) / Math.pow(10, decimal_places));
+            onChange?.(Number(input_value) / Math.pow(10, decimal_places));
+        }
```

Comment:
```suggestion
        const input = e.target.value.replace(/[^\d.]/g, '');
        const input_digits_length = input.replaceAll('.', '');
        const has_reached_max_digits = input_digits_length.length > max_digits;

        if (has_reached_max_digits) return;

        setValue(Number(input));
        onChange?.(Number(input));
```
@rostislav-deriv This wouldn't work? 🤔

Reply:
Committed the suggestion but the tests failed. Nevertheless, it is indeed better to rely not on `Math.pow` here, but on something simpler like string length; I'll try to make this work with this approach after resolving the locale issue and the cursor&pasting one 🙇🏻

Reply:
The locale makes this a bit harder btw, as the separator for the decimal points is not always `.`; need to account for this as well